### PR TITLE
[WIP] Fix test failures in CI

### PR DESCRIPTION
**Pull request by tmm1**
I'm trying to fix some tests are are currently failing. Each commit fixes a single test and the commit message contains the failing message from before the patch.

### CODE REVIEW

**Comment by tmm1** [comment]
Does it make sense to update the error message we're looking for in this test?

```
FAIL: tillegal_recursion.nim
Test "tests/objects/tillegal_recursion.nim" in category "objects"
Failure: reMsgsDiffer
Expected:
illegal recursion in type 'object'
Gotten:
inheritance only works with non-final objects
```

cc #1691

File: tests/method/tmapper.nim [hunk 1]
```diff
@@ -1,5 +1,5 @@
 discard """
-  errormsg: "invalid declaration order; cannot attach 'step' to method defined here: tmapper.nim(22,7)"
+  errormsg: "invalid declaration order; cannot attach 'step' to method defined here: tests/method/tmapper.nim(22,7)"
```
Comment (Araq) [inline tests/method/tmapper.nim]: Actually on Windows the path is just tmapper.nim(22,7). We need to get rid of this discrepancy.

File: lib/pure/ospaths.nim [hunk 2]
```diff
@@ -10,6 +10,10 @@
 # Included by the ``os`` module but a module in its own right for NimScript
 # support.
 
+when isMainModule:
```
Comment (Araq) [inline lib/pure/ospaths.nim]: Good one. :-)

File: tests/method/tmapper.nim (hunk 1 above)
Comment (tmm1) [inline tests/method/tmapper.nim]: Which one is correct?

Comment (Araq) [inline tests/method/tmapper.nim]: It doesn't matter but it needs to be consistent.

File: tests/objects/tillegal_recursion.nim [hunk 3]
```diff
@@ -1,5 +1,5 @@
 discard """
-  errormsg: "illegal recursion in type 'object'"
+  errormsg: "inheritance only works with non-final objects"
```
Comment (tmm1) [inline tests/objects/tillegal_recursion.nim]: Does this make sense?

**Comment by tmm1** [comment]
I'm not sure what to do about these two failures:

```
FAIL: tnot.nim
Test "tests/misc/tnot.nim" in category "misc"
Failure: reFilesDiffer
Expected:
tnot.nim
Gotten:
system.nim
```

```
FAIL: tenummix.nim
Test "tests/enum/tenummix.nim" in category "enum"
Failure: reFilesDiffer
Expected:
tenummix.nim
Gotten:
system.nim
```

They're both happening because errors inside templates are now expanded out into two lines:

```
tests/enum/tenummix.nim(11, 11) template/generic instantiation from here
lib/system.nim(351, 10) Error: type mismatch: got (TE1, TE2)
```

Should I update the assertion from looking for `tenummix.nim:11` to `system.nim:351`?

File: tests/objects/tillegal_recursion.nim (hunk 3 above)
Comment (Araq) [inline tests/objects/tillegal_recursion.nim]: Not really, but it's not too bad for now and makes the test green, so it's fine.

File: lib/pure/subexes.nim [hunk 4]
```diff
@@ -384,15 +384,15 @@ when isMainModule:
 
   doAssert "$#($', '10c'\n    '{#..})" % ["doAssert", "longishA", "longish"] ==
            """doAssert(
-    longishA,
+    longishA,
```
Comment (Araq) [inline lib/pure/subexes.nim]: That's fragile (we all hate trailing whitespace now), better use some smarter equality operation here.

**Comment by tmm1** [comment]
Only 4 failures remain on this branch. Two were mentioned above in https://github.com/nim-lang/Nim/pull/3403#issuecomment-144825148. The last two are a bit of a mystery and I've been unable to reproduce on my own ubuntu precise machine:

```
PASS: nimrtl.nim --app:lib -d:createNimRtl
PASS: server.nim --app:lib -d:useNimRtl
PASS: client.nim -d:useNimRtl
PASS: nimrtl.nim -d:release --app:lib -d:createNimRtl
PASS: server.nim -d:release --app:lib -d:useNimRtl
PASS: client.nim -d:release -d:useNimRtl
PASS: nimrtl.nim --gc:boehm --app:lib -d:createNimRtl
PASS: server.nim --gc:boehm --app:lib -d:useNimRtl
FAIL: client.nim --gc:boehm -d:useNimRtl
Test "tests/dll/client.nim" in category "dll"
Failure: reExitcodesDiffer
Expected:
exitcode: 0

Gotten:
exitcode: 1
Output:

PASS: nimrtl.nim -d:release --gc:boehm --app:lib -d:createNimRtl
PASS: server.nim -d:release --gc:boehm --app:lib -d:useNimRtl
FAIL: client.nim -d:release --gc:boehm -d:useNimRtl
Test "tests/dll/client.nim" in category "dll"
Failure: reExitcodesDiffer
Expected:
exitcode: 0

Gotten:
exitcode: 1
Output:

PASS: teffects1.nim
```

**Comment by Araq** [comment]
Any update on this? I'd love to merge it if it means travis runs our testsuite. :-)

**Comment by tmm1** [comment]
Sweet, 100% tests green!

**Pull request merged**